optuna_integration/comet/comet.py

```diff
@@ -0,0 +1,207 @@
+import functools
+import json
+from typing import Any
+from typing import Callable
+from typing import Dict
```

Comment:
Oning issue https://github.com/optuna/optuna/issues/4508 prefers `dict` instead of `typing.dict`. Similarly
`Union` -> `|`, and `Optional[SOME_TYPE]`-> `SOME_TYPE | None`.

https://github.com/optuna/optuna/issues/4508#issuecomment-1465563243